### DISCUSSION TITLE
How to query the highlight name for a char

### DISCUSSION THREAD

**Discussion by ofzo** [catppuccin/nvim#717] [Q&A] [answered]
In the rust project, the border of the popup is missing, but other projects are normal

rust project: 
<img width="750" alt="image" src="https://github.com/catppuccin/nvim/assets/18000037/009ddbf4-c2ef-435f-a4ce-25b12836ae0a">

cpp project:
<img width="551" alt="image" src="https://github.com/catppuccin/nvim/assets/18000037/f3cdf85e-be50-4e7a-9f52-43a7e528db0c">

ts project:
<img width="615" alt="image" src="https://github.com/catppuccin/nvim/assets/18000037/c3ce8377-5f0f-4af1-b43d-138084213564">

**Comment by ofzo** [comment]
my neovim version is 0.10.0

**Comment by vollowx** [comment]
If the problem is only seen in Rust projects, then it should be related to the Rust-related plugin you're using. Try to figure this plugin and configure its hover/preview window border or something.

**Comment by ilias777** [comment, marked as answer]
This is not a catppuccin thing. This has to be set in the `lsp-config` file.

You have to do this:

In your `lsp-config` file put this code:

```lua
local border = {
    { '┌', 'FloatBorder' },
    { '─', 'FloatBorder' },
    { '┐', 'FloatBorder' },
    { '│', 'FloatBorder' },
    { '┘', 'FloatBorder' },
    { '─', 'FloatBorder' },
    { '└', 'FloatBorder' },
    { '│', 'FloatBorder' },
}

local handlers = {
    ['textDocument/hover'] = vim.lsp.with(vim.lsp.handlers.hover, { border = border }),
    ['textDocument/signatureHelp'] = vim.lsp.with(vim.lsp.handlers.signature_help, { border = border }),
}
```

And then in the rust server initialization this:

```lua
local lspconfig = require('lspconfig')


lspconfig.rust_analyzer.setup({
    handlers = handlers,
})
```

And then you have the border for the popup.

<img width="1462" alt="border" src="https://github.com/catppuccin/nvim/assets/39212564/94647060-7360-4c47-ac6c-ca744e14e576">
